File: web/src/types/events.ts
// Types reflecting the backend /api/events schema (OpenAPI)
// ApiEvent: { event_type: 'info'|'warning'|'error', message: string, title: string, date: string (ISO datetime) }
export type ApiEventType = 'info' | 'warning' | 'error'

export interface ApiEvent {
  event_type: ApiEventType
  message: string
  title: string
  date: string
}

export interface EventApi {
  events: ApiEvent[]
}

File: src/meshmon/api/structure/events.py
import datetime
from enum import Enum

from pydantic import BaseModel


class ApiEventType(Enum):
    INFO = "info"
    WARNING = "warning"
    ERROR = "error"


class ApiEvent(BaseModel):
    event_type: ApiEventType
    message: str
    title: str
    date: datetime.datetime


class EventApi(BaseModel):
    events: list[ApiEvent] = []
